Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/jxbrowserutils/BrowserUtils.java

```diff
@@ -0,0 +1,273 @@
+
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.*;
+import com.microsoft.azure.hdinsight.common.HDInsightHelper;
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+import javafx.embed.swing.JFXPanel;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.math.BigInteger;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.FutureTask;
+
+import static java.lang.Thread.sleep;
+
+/**
+ * Created by t-rufang on 7/14/2017.
+ */
+class MyException extends Exception {
+    public MyException() {
+        super();
+    }
+
+    public MyException(String message) {
+        super(message);
+    }
+}
+
+class HDInsightHelperTest implements HDInsightHelper {
+
+    public void openJobViewEditor(@NotNull Object projectObject, @NotNull String uuid) {}
+    public void closeJobViewEditor(@NotNull Object projectObject, @NotNull String uuid) {}
+    public String getPluginRootPath() {
+        return ".";
+    }
+    public String getInstallationId() {
+        return "";
+    }
+    public boolean isIntelliJPlugin() {
+        return true;
+    }
+}
+
+
+public class BrowserUtils {
+    private static final ImmutableMap<String, String> map = ImmutableMap.of(
+            "win32", "jxbrowser-win32-6.16",
+            "win64", "jxbrowser-6.16",
+            "mac", "jxbrowser-mac-6.16",
+            "linux64", "jxbrowser-linux64-6.16",
+            "linux32", "jxbrowser-linux64-6.16");   // TODO: jxbrowser does't provide linux32 jar, we provide linux64 instead
+    public static final int MAX_DOWNLOAD_FAILURE_RETRY_TIMES = 3;
+    public static final int MAX_CHECK_FAILURE_RETRY_TIMES = 3;
+    public static final String storageConnectionString =
+            "DefaultEndpointsProtocol=https;"
+                    + "AccountName=ltiantest;"
+                    + "AccountKey=UBCWXXk1ha9ktwa5WTFN0MNSQOGMnyFg5KQvruM+AX4nQHxIQ7LjpTTIQtAM7ZsRbkhayUqy1AT80V+ZPssQFA==";
+    private static final Logger logger = LoggerFactory.getLogger(BrowserUtils.class);
+
+    public static boolean md5Check(String verifyFileName, String md5FileName) throws Exception {
```

Comment:
>String verifyFileName, String md5FileName [](start = 35, length = 41)

you can try to use hint @NotNull or @Nullable for all arguments.

Reply:
👍 I suggest it should be strict requirement for new adding codes.